### PR TITLE
Detect and handle CD-R discs

### PR DESCRIPTION
**Pull request by gorgobacka**
This PR uses cdrdao to detect CD-R discs as discussed in https://github.com/JoeLametta/whipper/issues/137 and stops the ripping process with a warning message. The parameter '--cdr' is required to rip this discs (similar to the '--unknown' parameter), indicating that the user should be sure that the inserted medium is reliable.

Additionally, the disc format is added to the output of 'Matching releases' to give the user more information.

~~Possible enhancements: add this information also to the logger.~~

### CODE REVIEW

**Comment by JoeLametta** [comment]
> Possible enhancements: add this information also to the logger.

Yep, apart from that it LGTM.

😉

**Comment by gorgobacka** [comment]
Thanks for your comment. I prepared an update.

**Comment by gorgobacka** [comment]
@JoeLametta I would like to know, what do you think about the my update. And I'm wondering, if there are any additional improvements needed before it can be merged.

**Comment by JoeLametta** [comment]
> I would like to know, what do you think about the my update. And I'm wondering, if there are any additional improvements needed before it can be merged.

I tested it yesterday and the feature seems to be working correctly. Here are the changes I'd like to be implemented before merging:

1. Rebase (there are conflicts)
2. PEP8 fix the PR (the whole codebase was recently made compliant)
3. Remove the MusicBrainz medium format thing (maybe propose it in a separate PR?)

Thanks,
Joe

**Comment by gorgobacka** [comment]
@JoeLametta Thanks for your fast reply. I will provide an update over the weekend.

File: whipper/command/cd.py [hunk 1]
```diff
@@ -143,6 +143,13 @@ def do(self):
                                 "--unknown not passed")
                 return -1
 
+        self.program.result.isCdr = cdrdao.DetectCdr(self.device)
+        if self.program.result.isCdr and \
```
Comment (JoeLametta) [inline whipper/command/cd.py]: I think it's better if you break it like this:

```python
        if (self.program.result.isCdr and
                not getattr(self.options, 'cdr', False)):
```

Comment (JoeLametta) [inline whipper/command/cd.py]: Everything else is OK

Comment (gorgobacka) [inline whipper/command/cd.py]: Indeed it looks better.

**Comment by JoeLametta** [comment]
Merged, thanks.

😉

**Pull request merged**